stix_shifter_modules/elastic_ecs/stix_translation/json/from_stix_map.json

```diff
@@ -55,5 +55,320 @@
+    }
+  },
+  "x-ecs-host": {
+    "fields": {
+      "architecture": ["host.architecture"],
+      "domain": ["host.domain"],
+      "hostname": ["host.hostname"],
+      "id": ["host.id"],
+      "ip": ["host.ip"],
+      "mac": ["host.mac"],
+      "name": ["host.name"],
+      "type": ["host.type"],
+      "uptime": ["host.uptime"]
+    }
+  },
+  "x-ecs-http": {
+    "fields": {
+      "request_body_bytes": ["http.request.body.bytes"],
+      "request_body_content": ["http.request.body.content"],
+      "request_bytes": ["http.request.bytes"],
+      "request_method": ["http.request.method"],
+      "request_referrer": ["http.request.referrer"],
+      "response_body_bytes": ["http.response.body.bytes"],
+      "response_body_content": ["http.response.body.content"],
+      "response_bytes": ["http.response.bytes"],
+      "response_status_code": ["http.response.method"],
+      "version": ["http.version"]
+    }
+  },
+  "x-ecs-log": {
+    "fields": {
+      "level": ["log.level"],
+      "logger": ["log.logger"],
+      "origin_file_line": ["log.origin.file.line"],
+      "origin_file_name": ["log.origin.file.name"],
+      "origin_function": ["log.origin.function"],
+      "original": ["log.original"],
+      "syslog_facility_code": ["log.syslog.facility.code"],
+      "syslog_facility_name": ["log.syslog.facility.name"],
+      "syslog_priority": ["log.syslog.priority"],
+      "severity_syslog_code": ["log.syslog.severity.code"],
+      "severity_syslog_name": ["log.syslog.severity.name"]
+    }
+  },
+  "x-ecs-observer": {
+    "fields": {
+      "egress_zone": ["observer.egress.zone"],
+      "egress_interface_alias": ["observer.egress.interface.alias"],
+      "egress_interface_id": ["observer.egress.interface.id"],
+      "egress_interface_name": ["observer.egress.interface.name"],
+      "hostname": ["observer.hostname"],
+      "ingress_zone": ["observer.ingress.zone"],
+      "ingress_interface_alias": ["observer.ingress.interface.alias"],
+      "ingress_interface_id": ["observer.ingress.interface.id"],
+      "ingress_interface_name": ["observer.ingress.interface.name"],
+      "ip": ["observer.ip"],
+      "mac": ["observer.mac"],
+      "name": ["observer.name"],
+      "product": ["observer.product"],
+      "serial_number": ["observer.serial_number"],
+      "type": ["observer.type"],
+      "vendor": ["observer.vendor"],
+      "version": ["observer.version"],
+      "os_name": ["observer.os.name"],
+      "os_platform": ["observer.os.platform"],
+      "os_version": ["observer.os.version"]
+    }
+  },
+  "x-ecs-organization": {
+    "fields": {
+      "id": ["organization.id"],
+      "name": ["organization.name"]
+    }
+  },
+  "x-ecs-pe": {
+    "fields": {
+      "company": ["pe.company"],
+      "description": ["pe.description"],
+      "file_version": ["pe.file_version"],
+      "original_file_name": ["pe.original_file_name"],
+      "product": ["pe.product"]
+    }
+  },
+  "x-ecs-related": {
```

Comment:
The ECS field guide indicates the values stored in the related properties will actually be stored in an array. Would we need to handle cases where a single value is getting passed into the pattern. It seems like `[x-ecs-related:ip = <some_ip>]` wouldn't work right now. The operators defined in the `query_constructor` have both = and IN mapped to ':' so I'm not even sure how to search inside of an array with an elastic query. Going the other way back into STIX, it's probably fine if we store the result as an array since it's a custom value.

Reply:
right. I found couple of other fields return values as an array. The problem is we don't have samples to test that out. But as we are just mapping those in custom attributes and not referencing that to any other object, it should be fine.